### PR TITLE
RFC proposal for `pdk config` subcommand

### PR DESCRIPTION
**Pull request by scotje**

### CODE REVIEW

File: RFCs/0000-add-pdk-config.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+### Proposed Configuration Keys
+
+#### User/Local Config
+
+The values for these config keys will be persisted in ~/.pdk/config. They are intended to be developer-specific settings
+and will not apply to another developer working on the same codebase.
+
+| Key | Description |
+| --- | --- |
+| user.module\_defaults.template.url | Default template URL to use when generating a new module. |
```
Comment (bmjen) [inline RFCs/0000-add-pdk-config.md]: `module_defaults.template.*`  values will also be used for `pdk convert`. To change these defaults, you would use `pdk config` again. Specifying `--template-url` will not change the defaults, as they currently do in the `answers.json`.

File: RFCs/0000-add-pdk-config.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+ - If your custom configuration is tool-specific, include a key and value that you can use to version your other
+   configuration data. That way you can significantly change the configuration format for your tool in the future
+   without breaking older versions. E.g. `module.x.super_module_tool.config_version = 1`
+
+ - Plus any other best practices we come up with.
+
+## Unresolved Questions
+
+- Do we need a “system”-level layer of config? (i.e. user-level config that is the default for all users of a system,
```
Comment (bmjen) [inline RFCs/0000-add-pdk-config.md]: -1 on the need for a system-level config.

Comment (vinzent) [reply RFCs/0000-add-pdk-config.md]: +1 for system level config 

Useful for being able to preset `*.module_defaults.template.*` settings to company defaults.

Comment (scotje) [reply RFCs/0000-add-pdk-config.md]: @vinzent Do you have multiple users using the same system for module development or would the "system" level config be provisioned automatically onto users workstations through Puppet, etc.?

Comment (vinzent) [reply RFCs/0000-add-pdk-config.md]: @scotje we have a central puppet-dev server where we develop puppet modules. nonetheless I would provision the setting by puppet. ;-)

File: RFCs/0000-add-pdk-config.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+- Resolve some inconsistencies and design challenges presented by existing PDK features. (See template-url/template-ref
+  design discussion and some issues we have had with invalid values cached in answers.json)
+
+- Improve the discoverability of module development configuration and customization. **Currently many of the avenues for
+  configuration are based around hidden files and their significance is not obvious to someone who is new to module
+  development.**
+
+## Proposed Implementation Design
+
+Implement a new pdk subcommand, `pdk config` which will offer the following actions:
```
Comment (ardrigh) [inline RFCs/0000-add-pdk-config.md]: Quick question: Is the intention here to have a configuration command and config file structure similar to `git config`?

The command structure looks similar but the RFC does not contain a sample configuration file example. An example file in the RFC might help.

File: RFCs/0000-add-pdk-config.md (hunk 2 above)
Comment (scotje) [reply RFCs/0000-add-pdk-config.md]: @vinzent Thanks for the additional context. Would lack of the system-level config in the initial release cause you to not adopt `pdk config` right away? Trying to narrow down whether that feature is a "must have" or a "nice to have" for the first release.

**Comment by scotje** [comment]
Going to propose moving this to FCP (Final Comment Period) during our PDK meeting on 25 July.

**Comment by scotje** [comment]
Moving to Final Comment Period which will end on August 3, 2018.

File: RFCs/0000-add-pdk-config.md (hunk 2 above)
Comment (scotje) [reply RFCs/0000-add-pdk-config.md]: I think we will defer system-level config for now and we can always add it in a subsequent release.

**Pull request merged**